js/util.bitcore.js

```diff
@@ -358,90 +315,74 @@ CWBitcore.signHash = function(signedTx, scriptPubKey, vin_index, cwPrivateKey, c
-  fnToSign[bitcore.Script.TX_PUBKEY]     = bitcore.TransactionBuilder.prototype._signPubKey;
-  fnToSign[bitcore.Script.TX_MULTISIG]   = bitcore.TransactionBuilder.prototype._signMultiSig;
-  fnToSign[bitcore.Script.TX_SCRIPTHASH] = bitcore.TransactionBuilder.prototype._signScriptHash;
-
-  // build key map
-  var wkMap = {};
-  wkMap[address] = new bitcore.WalletKey({network:NETWORK, privKey:cwPrivateKey.getBitcoreECKey()});
-
-  // unserialize raw transaction
-  var unsignedTx = CWBitcore.parseRawTransaction(unsignedHex);   
-
-  // prepare  signed transaction
-  var signedTx = new bitcore.TransactionBuilder();
-  //signedTx.tx = CWBitcore.prepareSignedTransaction(unsignedTx);
-  signedTx.tx = unsignedTx;
-
-  for (var i=0; i < unsignedTx.ins.length; i++) {
-      
-    // init parameters
-    var txin = unsignedTx.ins[i];
-    var scriptPubKey = new bitcore.Script(txin.s);
-    var input = {
-        address: address,
-        scriptPubKey: scriptPubKey,
-        scriptType: scriptPubKey.classify(),
-        i: i
-    };     
-    // generating hash for signature
-    var txSigHash = unsignedTx.hashForSignature(scriptPubKey, i, bitcore.Transaction.SIGHASH_ALL);
-    // empty the script
-    signedTx.tx.ins[i].s = bitcore.util.EMPTY_BUFFER;
-    // sign hash
-    var ret = fnToSign[input.scriptType].call(signedTx, wkMap, input, txSigHash);    
-    // inject signed script in transaction object
-    if (ret && ret.script) {
-      signedTx.tx.ins[i].s = ret.script;
-      if (ret.inputFullySigned) signedTx.inputsSigned++;
-      if (ret.signaturesAdded) signedTx.signaturesAdded += ret.signaturesAdded;
-    }
-
+  var tx = bitcore.Transaction(unsignedHex);
+
+  var keyMap = CWBitcore.genKeyMap([cwPrivateKey]);
+  var scriptPubKeys = {};
+  var keyChain = [];
+
+  var done = 0;
+  for (var idx = 0; idx < tx.inputs.length; idx++) {
+    (function(idx) {
+      var input = tx.inputs[idx];
+
+      failoverAPI('get_script_pub_key', {
+        tx_hash: input.prevTxId.toString('hex'),
+        vout_index: input.outputIndex
+      }, function (data) {
+        scriptPubKeys[idx] = bitcore.Script(data['scriptPubKey']['hex']);
+
+        if (++done == tx.inputs.length) {
+          var utxos = tx.inputs.map(function (index, idx) {
+            console.log(idx);
+            var scriptPubKey = scriptPubKeys[idx];
+
+            console.log(scriptPubKey);
+            console.log(scriptPubKey.toASM());
+            var address = scriptPubKey.toAddress(NETWORK);
+
+            console.log(address.toString());
+
+            if (address && typeof keyMap[address.toString()] !== "undefined") {
+              keyChain.push(keyMap[address.toString()]);
+
+              return {
+                txId: input.prevTxId.toString('hex'),
+                outputIndex: input.outputIndex,
+                script: scriptPubKey,
+                satoshis: bitcore.Unit.fromBTC(data['value']).toSatoshis()
```

Comment:
blindly asuming `get_script_pub_key` is going to return the same data as `getrawtransaction`